Kamel run -o incompatible with --dev

Running
```
kamel run ./examples/languages/Sample.java --dev -o yaml
```
Will produce an error like
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x1d9b80f]

goroutine 1 [running]:
github.com/apache/camel-k/pkg/util/watch.HandleIntegrationStateChanges(0x2748458, 0xc000064280, 0x27829a0, 0xc0007f86c0, 0x0, 0xc000bb34c8, 0x0, 0x0, 0x0)
	github.com/apache/camel-k/pkg/util/watch/watch.go:35 +0x8f
github.com/apache/camel-k/pkg/cmd.(*runCmdOptions).waitForIntegrationReady(0xc000af0e00, 0xc00081b900, 0x27829a0, 0xc0007f86c0, 0x0, 0xc0007f86c0, 0x0, 0xc000848010)
	github.com/apache/camel-k/pkg/cmd/run.go:416 +0x9e
github.com/apache/camel-k/pkg/cmd.(*runCmdOptions).run(0xc000af0e00, 0xc00081b900, 0xc00044b2c0, 0x1, 0x4, 0x0, 0x0)
	github.com/apache/camel-k/pkg/cmd/run.go:343 +0x4be
github.com/spf13/cobra.(*Command).execute(0xc00081b900, 0xc00044b280, 0x4, 0x4, 0xc00081b900, 0xc00044b280)
	github.com/spf13/cobra@v1.2.1/command.go:856 +0x472
github.com/spf13/cobra.(*Command).ExecuteC(0xc00081ac80, 0xc0007f9b80, 0xc00004c1e0, 0x6)
	github.com/spf13/cobra@v1.2.1/command.go:974 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.2.1/command.go:902
main.main()
	github.com/apache/camel-k/cmd/kamel/main.go:47 +0x17d
```
We probably need to include a check to warn the user the two options are not compatible.

--- a/pkg/cmd/run.go
+++ b/pkg/cmd/run.go
@@ -244,7 +244,7 @@ func (o *runCmdOptions) validate() error {
 	}
 
 	if o.OutputFormat != "" && o.Dev {
-		return errors.New("option '-o' is not compatible with '--dev'")
+		return fmt.Errorf("cannot use --dev with -o/--output option")
 	}
 
 	for _, label := range o.Labels {

--- a/pkg/cmd/run.go
+++ b/pkg/cmd/run.go
@@ -243,6 +243,10 @@ func (o *runCmdOptions) validate() error {
 		return err
 	}
 
+	if o.OutputFormat != "" && o.Dev {
+		return errors.New("option '-o' is not compatible with '--dev'")
+	}
+
 	for _, label := range o.Labels {
 		parts := strings.Split(label, "=")
 		if len(parts) != 2 {

--- a/pkg/cmd/run_test.go
+++ b/pkg/cmd/run_test.go
@@ -132,6 +132,16 @@ func TestRunDevFlag(t *testing.T) {
 	assert.Equal(t, true, runCmdOptions.Dev)
 }
 
+func TestRunDevModeOutputFlag(t *testing.T) {
+	runCmdOptions, rootCmd, _ := initializeRunCmdOptions(t)
+	_, err := test.ExecuteCommand(rootCmd, cmdRun, "--dev", "-o", "yaml", integrationSource)
+	assert.Equal(t, true, runCmdOptions.Dev)
+	assert.Equal(t, "yaml", runCmdOptions.OutputFormat)
+	assert.NotNil(t, err)
+	assert.Equal(t, "option '-o' is not compatible with '--dev'",
+		err.Error())
+}
+
 func TestRunEnvFlag(t *testing.T) {
 	runCmdOptions, rootCmd, _ := initializeRunCmdOptions(t)
 	_, err := test.ExecuteCommand(rootCmd, cmdRun,

--- a/pkg/cmd/run_test.go
+++ b/pkg/cmd/run_test.go
@@ -138,7 +138,7 @@ func TestRunDevModeOutputFlag(t *testing.T) {
 	assert.Equal(t, true, runCmdOptions.Dev)
 	assert.Equal(t, "yaml", runCmdOptions.OutputFormat)
 	assert.NotNil(t, err)
-	assert.Equal(t, "option '-o' is not compatible with '--dev'",
+	assert.Equal(t, "cannot use --dev with -o/--output option",
 		err.Error())
 }
 

